reflect.go

```diff
@@ -111,8 +112,20 @@ type Reflector struct {
 
 	// TypeMapper is a function that can be used to map custom Go types to jsconschema types.
 	TypeMapper func(reflect.Type) *Type
+
+	// NamingConvention specifies how Go field names should be converted to jsonschema property names.
+	NamingConvention NamingConvention
```

Comment:
This can just be a `func(string) string`, then we can drop the extra dependency.

Reply:
I like that approach, switched it to a closure.